azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java

```diff
@@ -174,13 +178,33 @@ public DispatchMethod getDispatchMethod(final ExecutableFlow flow) {
   }
 
   /**
-   * This method is used to start queue processor thread. The queue processor thread will pick
-   * execution from queue maintained in database and create a container for it.
+   * This is responsible for starting processing loops for the containerized dispatch.
+   * Currently this includes:
+   * (1) This method is used to start queue processor thread. The queue processor thread will pick
+   *   execution from queue maintained in database and create a container for it.
+   * (2) Starting container event watch, if present.
    */
   @Override
   public void start() {
     this.queueProcessor = setupQueueProcessor();
     this.queueProcessor.start();
+    this.startWatch();
+  }
+
+  // Start the event watch if configured.
+  private void startWatch() {
+    if (!containerizedWatch.isPresent()) {
+      logger.info("Containerized watch was not provided and will not be started");
+      return;
+    }
+    if (this.azkProps.getBoolean(ContainerizedDispatchManagerProperties.KUBERNETES_WATCH_ENABLED,
+        false) == true) {
+      logger.info("Starting containerized watch.");
+      this.containerizedWatch.get().launchWatch();
```

Comment:
No exception handling is required here?

Reply:
`ContainerizedWatch` and the underlying implementation in `KubernetesWatch` are designed to not throw any checked exceptions. They can throw unchecked exceptions though, however at this point in the code it seemed correct for those exceptions to be propagated to the callers of this method.